lib/iris/_concatenate.py

```diff
@@ -775,37 +775,21 @@ def _calculate_extents(self):
         self.dim_extents = []
         for coord, order in zip(self.dim_coords, self.dim_order):
             if order == _CONSTANT or order == _INCREASING:
-                points = _Extent(
-                    coord.core_points()[0], coord.core_points()[-1]
-                )
-                if coord.core_bounds() is not None:
+                points = _Extent(coord.points[0], coord.points[-1])
+                if coord.bounds is not None:
                     bounds = (
-                        _Extent(
-                            coord.core_bounds()[0, 0],
-                            coord.core_bounds()[-1, 0],
-                        ),
-                        _Extent(
-                            coord.core_bounds()[0, 1],
-                            coord.core_bounds()[-1, 1],
-                        ),
+                        _Extent(coord.bounds[0, 0], coord.bounds[-1, 0]),
+                        _Extent(coord.bounds[0, 1], coord.bounds[-1, 1]),
                     )
                 else:
                     bounds = None
             else:
                 # The order must be decreasing ...
-                points = _Extent(
-                    coord.core_points()[-1], coord.core_points()[0]
-                )
-                if coord.core_bounds() is not None:
+                points = _Extent(coord.points[-1], coord.points[0])
+                if coord.bounds is not None:
                     bounds = (
-                        _Extent(
-                            coord.core_bounds()[-1, 0],
-                            coord.core_bounds()[0, 0],
-                        ),
-                        _Extent(
-                            coord.core_bounds()[-1, 1],
-                            coord.core_bounds()[0, 1],
-                        ),
+                        _Extent(coord.bounds[-1, 0], coord.bounds[0, 0]),
+                        _Extent(coord.bounds[-1, 1], coord.bounds[0, 1]),
```

Comment:
Same applies here in `iris._concatenate._CoordSignature._calculate_extents`.

We're dealing exclusively with `DimCoord`s, therefore the `points` and `bounds` are **never** lazy and always realized.